src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -87,6 +87,37 @@ public void CreateContextCreatesNewContextIfNotUsingDefaultHttpContextFactory()
             hostingApplication.DisposeContext(context, null);
         }
 
+        [Fact]
+        public void IActivityFeatureIsCreated()
+        {
+            using var listener = new ActivityListener
+            {
+                ShouldListenTo = activitySource => (activitySource.Name == "Microsoft.AspNetCore" ||
+                                                    activitySource.Name == "DummySource"),
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData
+            };
+            ActivitySource.AddActivityListener(listener);
+
+            var hostingApplication = CreateApplication();
```

Comment:
This test can cause side effects. I've made this mistake before 😄 

I assume it's fine here since are you aren't doing anything on the `ActivityStarted`/`ActivityStopped` callback, but mentioning it in case you spot an issue.

Reply:
Explain?

Reply:
Gotcha, so I should use a unique name and pass that to the application

Reply:
You could add an overload to `CreateApplication` where you pass in your own ActivitySource. And then in your listener's `ShouldListenTo` callback, check for object equality (as opposed to string equality) in the ActivitySource name. This way your guaranteed not to have side effects and not cause side effects in other tests too.

Reply:
Using random name and ReferenceEquals